### PR TITLE
Add `es6.array.slice` in corejs2 builtins

### LINKED ISSUE #12859: Internal error in the corejs2 provider: unknown polyfill "es6.array.slice"
## Bug Report
after upgrade babel to 7.13.* I got this error if I use Array.silce in my code:
**Internal error in the corejs2 provider: unknown polyfill "es6.array.slice"**

**Current behavior**
image

**Input Code**

```js
data.slice(0, 20)
```

**Expected behavior**
it seems that es6.array.slice shouldn't be a polyfill

```
- Babel version(s): [e.g. v7.13.0]
- How you are using Babel: `webpack`

### PR DESCRIPTION
**Pull request by eligao**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #12859 
| Patch: Bug Fix          |
| Tests Added + Pass?      | Yes|
| Documentation PR Link    | 
| Any Dependency Changes?  | N/A |
| License                  | MIT

This polyfill  (es6.array.slice)[https://github.com/zloirock/core-js/blob/v2/modules/es6.array.slice.js]  is present in `corejs2` but missing in the `compat-data` package.
This introduces the error `Internal error in the corejs2 provider: unknown polyfill "es6.array.slice"`

### CODE REVIEW

File: packages/babel-compat-data/data/corejs2-built-ins.json [hunk 1]
```diff
@@ -850,6 +850,17 @@
     "samsung": "4",
     "electron": "0.30"
   },
+  "es6.array.slice": {
```
Comment (eligao) [inline packages/babel-compat-data/data/corejs2-built-ins.json:853]: comes from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice

**Comment by nicolo-ribaudo** [comment]
Thanks!

We use [this file](https://github.com/babel/babel/blob/main/packages/babel-compat-data/scripts/data/corejs2-built-in-features.js) to generate compat data from compat-table, could you update it?

**Comment by eligao** [comment]
> Build successful! You can test your changes in the REPL here: https://babeljs.io/repl/build/42103/

This REPL still reporting the same error, but this change fixed my local project builds already. Is the REPL supposed to load `@babel/compat-data` from this unpublished branch?

image

**Comment by nicolo-ribaudo** [comment]
The problem is that there is a cross-dependency between this repository and https://github.com/babel/babel-polyfills, so it won't be fixed in the REPL until we publish a Babel release.

**Comment by nicolo-ribaudo** [comment]
Thanks :+1:

One last thing: could you update the `babel-plugin-polyfill-corejs2` to 0.1.4 dependency `@babel/preset-env` and `@babel/transform-runtime`? This makes it sure that we rely on a version that never throws the error.

File: packages/babel-compat-data/scripts/data/corejs2-built-in-features.js [hunk 2]
```diff
@@ -73,8 +73,8 @@ module.exports = {
   "es6.array.of": "Array static methods / Array.of",
   "es6.array.reduce": "Array methods / Array.prototype.reduce",
   "es6.array.reduce-right": "Array methods / Array.prototype.reduceRight",
-  // compat-table missing babel6 mapping
```
Comment (eligao) [inline packages/babel-compat-data/scripts/data/corejs2-built-in-features.js:76]: Is this still a valid reason? I'm seeing a few more `compat-table missing babel6 mapping` here

Comment (nicolo-ribaudo) [reply packages/babel-compat-data/scripts/data/corejs2-built-in-features.js:76]: Uh no, we can add all of them back if they are in compat-table.

Comment (eligao) [reply packages/babel-compat-data/scripts/data/corejs2-built-in-features.js:76]: Noted, let me add them back then

**Review by nicolo-ribaudo — APPROVED** [review]
Thanks!

I'm having lunch now, I'll release right after (so that CI has time to run)

**Comment by nicolo-ribaudo** [comment]
Whops, try running `yarn dedupe` to fix the CI error

**Comment by eligao** [comment]
> Thanks!
> 
> I'm having lunch now, I'll release right after (so that CI has time to run)

out for dinner too, be back later

**Comment by clarkdo** [comment]
@nicolo-ribaudo @eligao We found another similar error `Internal error in the corejs2 provider: unknown polyfill "web.dom.iterable"` when using `@babel/preset-env` with `targets: {node: 'current'}` and `corejs v2`.

It seemed `web.dom.iterable` is not in `corejs-built-ins` and [only added for web targets](https://github.com/babel/babel-polyfills/blob/092e66f649645d164305e282863c68d2d16f28ff/packages/babel-plugin-polyfill-corejs2/src/add-platform-specific-polyfills.js#L8)

**Comment by nicolo-ribaudo** [comment]
@clarkdo I made it a warning rather than an error in `@babel/helper-define-polyfill-provider@0.1.2`: we'll still need to fix it, but at least it doesn't fail your build.

**Comment by clarkdo** [comment]
> we'll still need to fix it, but at least it doesn't fail your build.

Thanks for the reply, actually building failed because `isRequired` is called after the warning which will use [compatData[name][target]](https://github.com/babel/babel/blob/main/packages/babel-helper-compilation-targets/src/filter-items.js#L75)(`polyfills['web.dom.iterable'].node` in my case)

**Comment by nicolo-ribaudo** [comment]
Oh ok :upside_down_face: 

I can work on a PR later, unless you want to work on it.

**Comment by clarkdo** [comment]
@nicolo-ribaudo  I'm OK to take it, Do you think we should add it to `@babel/compat-data` or [platform-specific-polyfill](https://github.com/babel/babel-polyfills/blob/main/packages/babel-plugin-polyfill-corejs2/src/add-platform-specific-polyfills.js) ?

**Comment by nicolo-ribaudo** [comment]
The fix should be in `babel-plugin-polyfill-corejs2`. Looking at the code, I think it would be enough to use [`injectIfAvailable`](https://github.com/babel/babel-polyfills/blob/main/packages/babel-plugin-polyfill-corejs2/src/index.js#L72) instead of `inject`.

However, `injectIfAvailable` should probably still throw (or warn) if a polyfill is not available and it's not one of the "possibly unavailable polyfills" defined in https://github.com/babel/babel-polyfills/blob/55f582c8ebc4b8a181a51fecaa92c89158cc7ac9/packages/babel-plugin-polyfill-corejs2/src/add-platform-specific-polyfills.js#L5-L15.

**Comment by clarkdo** [comment]
Agree, I think we'd better cover the case in which `web.dom.iterable` is missing.

I can open a pr tmr morning for using injectIfAvailable instead of inject, then we can discuss about better fix.

**Comment by nicolo-ribaudo** [comment]
Ok thanks!

**Pull request merged**